### ISSUE TITLE
Defining unions (RFC 1444)?

### ISSUE THREAD

**Issue by kamalmarhubi** [rust-lang/libc#303]
[RFC 1444](https://github.com/rust-lang/rfcs/blob/master/text/1444-union.md) was accepted a few weeks ago, bringing us actual real support for unions. Will libc grow to support for unions in its struct definitions?

The big issue I see for it is the RFC doesn't add unnamed unions, which covers most if not all of the unions we have in libc. At the same time, it'd be really great to not have to define the unions in client code, and to have libc's stronger testing applied to them.

The other issue I see is Rust version compatibility. I'll open another issue for that.

**Comment by alexcrichton** [comment]
Certainly! We may not be able to change existing types but we can certainly add more! This is mostly just covered by the version policy, so I'm gonna close in favor of that.

(also just waiting on unions to get implemented...)

**Comment by kamalmarhubi** [comment]
> We may not be able to change existing types

I'd argue for a semver incompatible bump to include proper definitions, even for existing types. Depending on how various bitfield RFCs are going, this bump could perhaps wait to encompass both.

**Comment by alexcrichton** [comment]
Ah yeah that'd also work, but a semver bump of libc is not to be taken lightly, we wouldn't want to do it just for that.
